website/addons/base/views.py

```diff
@@ -548,19 +585,21 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
 
     if version is None:
         if file_node.get_guid():
-            # If this file has been successfully view before but no longer exists
-
-            # Move file to trashed file node
+            # If this file has been successfully view before, or OSF, but no
```

Comment:
What does "or OSF" mean? Also if you're updating this already you change "view" to "viewed"?